### PR TITLE
Feature securitygroup create

### PR DESCRIPTION
**Pull request by varikin**

### CODE REVIEW

**Comment by drmaas** [comment]
+1

when we do the PR to upstream, may find that veritas also added the OpenstackClientProvider.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy [hunk 1]
```diff
@@ -52,18 +58,65 @@ abstract class OpenstackClientProvider {
...
+   * @param securityGroupId id of an existing security group to update
+   * @param securityGroupName name security group
+   * @param description description of the security group
+   * @param rules list of rules for the security group
+   */
+  void upsertSecurityGroup(String securityGroupId, String securityGroupName, String description, List<OpenstackSecurityGroupDescription.Rule> rules) {
+
+     handleRequest(AtomicOperations.UPSERT_SECURITY_GROUP) {
+
+       // The call to getClient reauthentictes via a token, so grab once for this method
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: Global comment ... I would remove comments that don't provide value.  If the code is straight forward, then no need for comments.  If the code is complex, then comments are required.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/securitygroup/UpsertOpenstackSecurityGroupAtomicOperationConverter.groovy [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import com.netflix.spinnaker.clouddriver.openstack.deploy.converters.OpenstackAtomicOperationConverterHelper
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.securitygroup.OpenstackSecurityGroupDescription
+import com.netflix.spinnaker.clouddriver.openstack.deploy.ops.securitygroup.UpsertOpenstackSecurityGroupAtomicOperation
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
+import com.netflix.spinnaker.clouddriver.security.AbstractAtomicOperationsCredentialsSupport
+import org.springframework.stereotype.Component
+
+/**
+ *
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/securitygroup/UpsertOpenstackSecurityGroupAtomicOperationConverter.groovy]: Remove empty comment block

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/exception/OpenstackOperationException.groovy [hunk 3]
```diff
@@ -25,11 +25,15 @@ class OpenstackOperationException extends RuntimeException {
     super("$operation failed: ${e.message}".toString(), e)
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/exception/OpenstackOperationException.groovy]: Can we remove the toString() here too?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy (hunk 1 above)
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: I was pointing out that `getClient` is not a simple getter, so I only wanted to call it once for this thread.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/securitygroup/UpsertOpenstackSecurityGroupAtomicOperationConverter.groovy (hunk 2 above)
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/converters/securitygroup/UpsertOpenstackSecurityGroupAtomicOperationConverter.groovy]: done

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/exception/OpenstackOperationException.groovy (hunk 3 above)
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/exception/OpenstackOperationException.groovy]: Done

**Comment by varikin** [comment]
Closing and re-targeting to Spinnaker upstream.

**Pull request closed without merging**